lib/postgres.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Library made to simplify accessing / connecting to postgres databases,
+ * and to cleanly handle when the pg module isn't installed.
+ * @author mia-pi-git
+ */
+
+// @ts-ignore in case module doesn't exist
+import type * as PG from 'pg';
+import type {SQLStatement} from 'sql-template-strings';
+import * as Streams from './streams';
+
+export class PostgresDatabase {
+	private pool: PG.Pool;
+	constructor(config = PostgresDatabase.getConfig()) {
+		try {
+			this.pool = new (require('pg').Pool)(config);
+		} catch (e) {
+			this.pool = null!;
+		}
+	}
+	async query(statement: string | SQLStatement, values?: any[]) {
+		if (!this.pool) {
+			throw new Error(`Attempting to use postgres without 'pg' installed`);
+		}
+		let result;
+		try {
+			result = await this.pool.query(statement, values);
+		} catch (e) {
+			// postgres won't give accurate stacks unless we do this
+			throw new Error(e.message);
+		}
+		return result?.rows || [];
+	}
+	static getConfig() {
+		let config: AnyObject = {};
+		try {
+			config = require('../config/config').usepostgres;
+			if (!config) throw new Error('Missing config for pg database');
+		} catch (e) {}
+		return config;
+	}
+	async transaction(callback: (conn: PG.PoolClient) => any, depth = 0): Promise<any> {
+		const conn = await this.pool.connect();
+		await conn.query(`BEGIN`);
+		let result;
+		try {
+			// eslint-disable-next-line callback-return
+			result = await callback(conn);
+		} catch (e) {
```

Comment:
```suggestion
		} catch (e: any) {
```
This goes for the other catch statements also.

Reply:
Done.